### PR TITLE
Refactor the conflict error functions to ConflictHandler

### PR DESCRIPTION
**Pull request by nan-yu**
It also unifies FightHandler to use the orderedMap.

### CODE REVIEW

File: pkg/remediator/conflict/handler.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+func (h *handler) AddConflictError(e status.ManagementConflictError) {
+	h.mux.Lock()
+	defer h.mux.Unlock()
+
+	for _, existingErr := range h.conflictErrs {
+		if e.Error() == existingErr.Error() {
+			return
+		}
+	}
+	h.conflictErrs = append(h.conflictErrs, e)
```
Comment (karlkfi) [inline pkg/remediator/conflict/handler.go:60]: Should we sort on add so they stay in a consistent order? Or are they always just ordered by order of appearance?

File: pkg/remediator/watch/filteredwatcher.go [hunk 2]
```diff
@@ -109,17 +109,16 @@ var _ Runnable = &filteredWatcher{}
...
-		removeConflictErrorFunc: cfg.removeConflictErrorFunc,
+		gvk:             cfg.gvk.String(),
+		startWatch:      cfg.startWatch,
+		resources:       cfg.resources,
+		queue:           cfg.queue,
+		scope:           cfg.scope,
+		syncName:        cfg.syncName,
+		base:            watch.NewEmptyWatch(),
+		errorTracker:    make(map[string]time.Time),
+		conflictErrMap:  make(map[queue.GVKNN]status.ManagementConflictError),
```
Comment (karlkfi) [inline pkg/remediator/watch/filteredwatcher.go:120]: Is this map something the ConflictHandler could handle? Does it need to be locked the same way?

Comment (karlkfi) [reply pkg/remediator/watch/filteredwatcher.go:120]: Seems redundant to store as both a map and a list, unless we care about the performance at scale. If performance and order are important, we could used an ordered map, like https://pkg.go.dev/github.com/wk8/go-ordered-map

File: pkg/remediator/conflict/handler.go (hunk 1 above)
Comment (nan-yu) [reply pkg/remediator/conflict/handler.go:60]: What about sorting on get?

Comment (nan-yu) [reply pkg/remediator/conflict/handler.go]: Switched to use orderedmap.

File: pkg/remediator/watch/filteredwatcher.go (hunk 2 above)
Comment (nan-yu) [reply pkg/remediator/watch/filteredwatcher.go]: Switched to use ordered map.

**Review by karlkfi — APPROVED** [review]
/lgtm

Looks much better, thanks!

**Pull request merged**